src/Interval.hs

```diff
@@ -69,94 +69,94 @@ interval <-> x =  interval <+> (-x)
 
 
 raisePerfect :: IQuality -> IQuality
-raisePerfect IPerfect = (IAugmented 1)
-raisePerfect (IAugmented x) = (IAugmented (x+1))
+raisePerfect IPerfect = IAugmented 1
+raisePerfect (IAugmented x) = IAugmented $ x + 1
 raisePerfect (IDiminished 1) = IPerfect
-raisePerfect (IDiminished x) = (IDiminished (x-1)) 
+raisePerfect (IDiminished x) = IDiminished $ x - 1
 
 raiseMajor :: IQuality -> IQuality
-raiseMajor IMajor = (IAugmented 1)
-raiseMajor (IAugmented x) = (IAugmented (x+1))
+raiseMajor IMajor = IAugmented 1
+raiseMajor (IAugmented x) = IAugmented $ x + 1
 raiseMajor IMinor = IMajor
 raiseMajor (IDiminished 1) = IMinor
-raiseMajor (IDiminished x) = (IDiminished (x-1))
+raiseMajor (IDiminished x) = IDiminished $ x - 1
 
 lowerPerfect :: IQuality -> IQuality
-lowerPerfect IPerfect = (IDiminished 1)
-lowerPerfect (IDiminished x) = (IDiminished (x+1)) 
+lowerPerfect IPerfect = IDiminished 1
+lowerPerfect (IDiminished x) = IDiminished $ x + 1
 lowerPerfect (IAugmented 1) = IPerfect
-lowerPerfect (IAugmented x) = (IAugmented (x-1))
+lowerPerfect (IAugmented x) = IAugmented $ x-1
 
 lowerMajor :: IQuality -> IQuality
 lowerMajor IMajor = IMinor
-lowerMajor IMinor = (IDiminished 1)
-lowerMajor (IDiminished x) = (IDiminished (x+1))
+lowerMajor IMinor = IDiminished 1
+lowerMajor (IDiminished x) = IDiminished $ x + 1
 lowerMajor (IAugmented 1) = IMajor
-lowerMajor (IAugmented x) = (IAugmented (x-1))
+lowerMajor (IAugmented x) = IAugmented $ x - 1
 
 
 defaultIQuality :: Int -> IQuality
 defaultIQuality i =
-  case (intMod i) of
+  case intMod i of
     intervalInt
       | intervalInt `elem` [1, 4, 5] -> IPerfect
       | intervalInt `elem` [2, 3, 6, 7] -> IMajor
       | otherwise -> error "Impl error, mod 7 issue"
- 
+
 
 modShift :: Int -> Int -> (Int -> Int)
-modShift newZero modulus = (\i -> ((i - newZero) `mod` modulus) + newZero)
+modShift newZero modulus i = ((i - newZero) `mod` modulus) + newZero
 
 intMod :: Int -> Int
 intMod = modShift 1 7
 
-intervalMod :: Interval -> Interval 
-intervalMod (Interval iQual i) = Interval iQual (intMod i)
+intervalMod :: Interval -> Interval
+intervalMod (Interval iQual i) = Interval iQual $ intMod i
 
 lowestAbsValue :: Int -> Int
 lowestAbsValue = modShift (-6) 12
 
 
 invert :: Interval -> Interval
-invert (Interval iQual i) = 
-  let 
-    newI = intMod $ 9 - (intMod i)
-    newQual = 
-      case iQual of 
+invert (Interval iQual i) =
+  let
+    newI = intMod $ 9 - intMod i
+    newQual =
+      case iQual of
         IMajor -> IMinor
         IMinor -> IMajor
         IPerfect -> IPerfect
-        (IAugmented x) -> (IDiminished x)
-        (IDiminished x) -> (IAugmented x)
-  in 
+        (IAugmented x) -> IDiminished x
+        (IDiminished x) -> IAugmented x
+  in
     Interval newQual newI
 
 
-infixl 6 |+| 
-(|+|) i1 i2 = intervalAdd i1 i2
+infixl 6 |+|
+(|+|) = intervalAdd
```

Comment:
This is a common 'mistake' in Haskell code; we want the name `(|+|)` to mean the same thing as the name `intervalAdd`, so we should just say that directly; don't name things when you don't need to!

Reply:
To be even more concrete, this is an _eta-reduction_.